src/aspire/utils/rotation.py

```diff
@@ -0,0 +1,280 @@
+"""
+Define a Rotation Class for customized rotation operations used by ASPIRE.
+"""
+
+import copy
+
+import numpy as np
+from numpy.linalg import norm
+from scipy.linalg import svd
+from scipy.spatial.transform import Rotation as sp_rot
+
+from aspire.utils import ensure
+from aspire.utils.random import Random
+
+
+class Rotation:
+    def __init__(self, num_rots, seed=None, seq="ZYZ", angles=None, dtype=np.float32):
```

Comment:
It's not a good idea to have the default constructor generate random rotations. It would be better for it to accept angles or rotation matrices and then have a static method that initializes a `Rotation` object with uniformly distributed random rotations.

Reply:
Changed to initialize by angles or matrices in commit 0c8f0b8.

Reply:
Pick one: initialize from angles *or* matrices. Since a lot of the interface suggests that this is a matrix-like object, it would make more sense to initialize from matrices.

Also, it still generates random angles if neither are supplied. Remove this.

Reply:
In real case we might have three major applications. 1) Generate Rotation object for Simulation class using random angles:  2) Generate Rotation object using rotational angles from STAR file; 3) Generate Rotation object for the rotation matrices predicted from orientation estimation. The constructor tries to cover these three cases. The using of angles is convenient so that no transform outside the class is needed?

Reply:
The constructor should do *one thing*. I suggest that one thing should be “build a rotation stack from an array of rotation matrices.” If we want other ways to construct a rotation stack, we use the static methods. Note that this is why the SciPy rotation object creation looks the way it does (there you aren't even supposed to use the constructor).

Reply:
Remove `angles` from the constructor and use `_matrices` to hold rotation stack in new commit f7fcc5cf.

Reply:
This is still generating random rotations if no matrices are supplied. The signature should be
```
def __init__(self, matrices)
```
that's it. All other ways to generate a Rotation object should be through the static methods.

Reply:
Done in new commit 51beb74.